### PR TITLE
[SNC-392] compile support for policy test command

### PR DESCRIPTION
**Pull request by davidmdm**
# Checklist

=========

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked for similar issues and haven't found anything relevant.
- [x] This is not a security issue (which should be reported here: https://circleci.com/security/)
- [x] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/main/CONTRIBUTING.md).

### Internal Checklist
- [x] I am requesting a review from my own team as well as the owning team
- [x] I have a plan in place for the monitoring of the changes that I am making (this can include new monitors, logs to be aware of, etc...)

## Changes

=======

- Updated the CircleCI-Public/circle-policy-agent
- Modified the `circleci policy test` command by adding the policy-agent compile function
  - calls the orbs-service API via graphql to compile test input
- Fixes a bug in the policy API client that was causing failures across tests
- adds the compily policy and test to validate the compilation is working in e2e fashion

## Rationale

=========

To provide a better testing experience of CircleCI Config Policies, we needed to update the policy-agent that understood the compile and pipeline_parameters directive, and provide a compilation function for it to use.

### CODE REVIEW

**Review by jbialy — APPROVED** [review]
From a high level it lgtm!

File: cmd/policy/policy_test.go [hunk 1]
```diff
@@ -1330,13 +1331,31 @@ func TestTestRunner(t *testing.T) {
 				assert.Contains(t, s, "<?xml")
 			},
 		},
+		{
```
Comment (sagar-connect) [inline cmd/policy/policy_test.go:1334]: Test suggestion:
`compile: true`, but owner-id isn't provided.

**Review by sagar-connect — APPROVED** [review]

File: cmd/policy/policy_test.go (hunk 1 above)
Comment (davidmdm) [reply cmd/policy/policy_test.go:1334]: That is the current test case

**Review by rooneyshuman — APPROVED** [review]

File: cmd/policy/policy_test.go (hunk 1 above)
Comment (sagar-connect) [reply cmd/policy/policy_test.go:1334]: My understanding is if compilation is required, but owner-id isn't provided, it should fail in some way.
But the expectation is pass in this test?

Comment (davidmdm) [reply cmd/policy/policy_test.go:1334]: Owner-ID is only required if you need to pull a private orb

**Review by JulesFaucherre — APPROVED** [review]

**Pull request merged**